detection_rules/action.py

```diff
@@ -0,0 +1,62 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
```

Comment:
Will you update the current `actions: Optional[list]` field with these classes?

Reply:
the idea we have discussed to this point was keeping these decoupled for smoother VCS. I think a smart compromise would be to `Optional[Union[list, Action]` or similar